### PR TITLE
Refactor Google Group checker to move to play-googleauth

### PR DESCRIPTION
**Pull request by dominickendrick**
I've refactored the google group checker within this branch to confirm the implementation is sane before migrating this to play-googleauth.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -11,7 +11,7 @@ object Dependencies {
   val identityTestUsers = "com.gu" %% "identity-test-users" % "0.5"
   val scalaUri = "com.netaporter" %% "scala-uri" % "0.4.6"
   val membershipCommon = "com.gu" %% "membership-common" % "0.63"
-  val playGoogleAuth = "com.gu" %% "play-googleauth" % "0.1.11"
+  val playGoogleAuth = "com.gu" %% "play-googleauth" % "0.1.12-SNAPSHOT"
```
Comment (dominickendrick) [inline project/Dependencies.scala]: This is awaiting the publishing of the play-googleauth package

**Comment by mattandrews** [comment]
Needs merge!

**Comment by jennysivapalan** [comment]
build fail :crying_cat_face:

**Comment by rtyley** [comment]
:+1:

**Pull request merged**